File: backend/internal/models/team.go
package models

type Team struct {
	DbModel
	CorrelationId string  `json:"correlation_id" validate:"required"`
	Name          string  `json:"name" validate:"required"`
	LogoId        string  `json:"logo_id" validate:"required"`
	Color         string  `json:"color" validate:"required"`
	LeagueID      uint    `json:"league_id" validate:"required"`
	League        *League  `json:"league"`
	Roster        *Roster `json:"roster"`
	RosterID      *uint   `json:"roster_id" validate:"required"`

	Wins   int `json:"wins" validate:"required"`
	Losses int `json:"losses" validate:"required"`
}

File: app/src/models/Team.ts
import type { DbModel, League, Roster } from './index'
import type { LeagueId, LogoId, RosterId, TeamGuid } from './ids'

export interface Team extends DbModel {
  correlation_id: TeamGuid
  name: string
  logo_id: LogoId
  color: string
  league_id: LeagueId
  league: League
  roster?: Roster
  roster_id?: RosterId
  wins: number
  losses: number
}

export type { TeamId, TeamGuid } from './ids'
